docs/sources/panels-visualizations/visualizations/flame-graph/index.md

```diff
@@ -0,0 +1,90 @@
+
+This process can be repeated to focus on finding the amount of resources that particular blocks and their children are consuming.
+
+{{< figure src="/static/img/docs/flame-graph-panel/flame-graph-search.png" class="docs-image--no-shadow" max-width= "900px" caption="Figure 4 - Search" >}}
+
+When you would like to quickly find the blocks with particular labels you can use the search field. Simply type in the label you want to search for and all of the blocks in the table with that text in their label will remain colored with all of the remaining blocks grayed out.
+
+You can also click on any block or hover over it to narrow your search.
+
+## Top table
+
+The top table allows you to visualize which symbols take up the most resources in your profile.
+
+The table has three columns, symbols, self and total - representing the self time and total time of each symbol. The table is sorted by self time by default, but can be reordered by total time or symbol name upon clicking the column headers.
+
+{{< figure src="/static/img/docs/flame-graph-panel/top-table.png" class="docs-image--no-shadow" max-width= "900px" caption="Figure 5 - Top table" >}}
+
+Clicking on a particular row of the top table will add that row's symbol to the search input which will search for that symbol in the flame graph.
+
+Clicking on that top table row again will remove its symbol from the search input.
+
+{{< figure src="/static/img/docs/flame-graph-panel/top-table-clicking.png" class="docs-image--no-shadow" max-width= "900px" caption="Figure 6 - Clicking on a top table row" >}}
+
+Of course, sometimes you will not need to view both the top table and flame graph at the same time. This is especially true for large profiles.
+
+In this case and if you only want to see the top table, you can click on the top table button to show only the top table.
+
+3 options are available:
+
+- Top Table: Only show the top table
+- Flame Graph: Only show the flame graph
+- Both: Show both the top table and flame graph
+
+{{< figure src="/static/img/docs/flame-graph-panel/selected-view.png" class="docs-image--no-shadow" max-width= "900px" caption="Figure 7 - Show the top table / flame graph / both" >}}
+
+## Data API
+
+The flame graph expects a data frame data in a particular format to be able to render it. As the data is hierarchical while a data frame has a tabular structure, the way to encode it is to use a [nested set model](https://en.wikipedia.org/wiki/Nested_set_model).
+
+In a simplified form it means each item of the flame graph is encoded just by its nesting level as an integer value, its metadata and by its order in the data frame. This means that the order of items is significant and needs to be correct. The ordering is depth first traversal of the items in the flame graph. This allows us to recreate the graph without needing variable length values in the data frame like children's array.
+
+Required fields:
```

Comment:
Can we introduce this table?

Reply:
I'm not sure what you mean? There is already a table below this

Reply:
I assume in the sense of giving it some description or why it's there? The whole section though is about what data shape does the flame graph require so I am not sure what would make sense here. Or do you mean just short `These are the required fields for flame graph:`?